### PR TITLE
Add reconfiguration methods to LegacyComponentSerializer.Builder

### PR DESCRIPTION
**Pull request by A248**
Adds methods to LegacyComponentSerializer builder to undo options already set on it.

### CODE REVIEW

File: text-serializer-legacy/src/main/java/net/kyori/adventure/text/serializer/legacy/LegacyComponentSerializer.java [hunk 1]
```diff
@@ -242,6 +261,26 @@
      */
     @NonNull Builder useUnusualXRepeatedCharacterHexFormat();
```
Comment (kashike) [inline text-serializer-legacy/src/main/java/net/kyori/adventure/text/serializer/legacy/LegacyComponentSerializer.java:262]:
```suggestion
    default @NonNull Builder useUnusualXRepeatedCharacterHexFormat() {
      return this.useUnusualXRepeatedCharacterHexFormat(true);
    }
```

File: text-serializer-legacy/src/main/java/net/kyori/adventure/text/serializer/legacy/LegacyComponentSerializer.java [hunk 2]
```diff
@@ -242,6 +261,26 @@
...
+     * the Spigot API which expect legacy strings in this format.</p>
+     *
+     * <p>It is recommended to use only when absolutely necessary, and when no better alternatives
+     * are available.</p>
+     *
+     * @param flag {@code true} to use the '&amp;x' repeated code format, {@code false} not to
+     * @return this builder
+     * @since 4.8.0
+     */
+    @NonNull Builder useUnusualXRepeatedCharacterHexFormat(boolean flag);
```
Comment (kashike) [inline text-serializer-legacy/src/main/java/net/kyori/adventure/text/serializer/legacy/LegacyComponentSerializer.java:282]:
```suggestion
    @NonNull Builder useUnusualXRepeatedCharacterHexFormat(final boolean flag);
```

File: text-serializer-legacy/src/main/java/net/kyori/adventure/text/serializer/legacy/LegacyComponentSerializer.java [hunk 3]
```diff
@@ -223,6 +231,17 @@
...
+    /**
+     * Sets whether the serializer should support hex colors.
+     *
+     * <p>If {@code false}, hex colors are downsampled to the nearest named color.</p>
+     *
+     * @param flag {@code true} to support hex colors, {@code false} not to
+     * @return this builder
+     * @since 4.8.0
+     */
+    @NonNull Builder hexColors(boolean flag);
```
Comment (kashike) [inline text-serializer-legacy/src/main/java/net/kyori/adventure/text/serializer/legacy/LegacyComponentSerializer.java:243]:
```suggestion
    @NonNull Builder hexColors(final boolean flag);
```

**Review by kashike — CHANGES_REQUESTED** [review]

**Review by lucko — APPROVED** [review]

**Review by kezz — APPROVED** [review]
`@since` tags need updating before merging.

**Comment by kezz** [comment]
@A248 do you plan to update this PR with the requested changes to since tags and annotations?

**Comment by A248** [comment]
I have no plans to do such things myself. Anyone else is free to continue this PR where I left off.

**Pull request closed without merging**